torch_xla/runtime.py

```diff
@@ -12,10 +12,41 @@
 import torch_xla.utils.utils as xu
 import torch_xla._internal.tpu as tpu
 from torch_xla.experimental import plugins
+from torch_xla import runtime
 
 R = TypeVar('R')
 FN = TypeVar('FN')
 
+# Note [Dynamo WORLD_SIEZ and ORDINAL]
+# Belows are workaround to cache the ordinal and world_size such that
+# Dynamo won't do graph breaks when runtime.xrt_world_size() and runtime.global_ordinal() are called.
+_WORLD_SIZE = None
+_ORDINAL = None
+
+
+def _init_world_size_ordinal():
+  global _WORLD_SIZE, _ORDINAL
+
+  # Dynamo doesn't support XRT or multithreaded runtime. See Note [V3-8 Threading]
+  if not runtime.using_pjrt() or runtime.addressable_device_count() > 1:
+    return
+
+  if _WORLD_SIZE is None:
+    _WORLD_SIZE = runtime.world_size()
+    _ORDINAL = runtime.global_ordinal()
+
+
+def run_once(func):
+  result = []
```

Comment:
nit: please use a `nonlocal` value instead of a list here https://stackoverflow.com/a/1261961

I just learned about this too :smile: